models/Import/ImportServiceProvider.php

```diff
@@ -95,10 +94,11 @@ public function __invoke(ContainerConfigurator $configurator): void
             ->public()
             ->args(
                 [
-
                     service(QtiRunnerService::SERVICE_ID),
                     service(DeliveryExecutionService::SERVICE_ID),
-                    service(DeliveryContainerService::SERVICE_ID),
+                    class_exists('oat\taoDeliveryRdf\model\DeliveryContainerService') ?
```

Comment:
As I understood your intent here, please correct me if I'm wrong, you are trying to decouple outcome extension from delivery-rdf extension.

So a couple of things here:
1. Unfortunately, service definition requires this third parameter no matter what, so if we put null there it will fail when it will try to instantiate a service. Please check the third parameter [here](https://github.com/oat-sa/extension-tao-outcome/blob/bbb231b43bb0ac052de3586fb7520649bb008110/models/Import/Service/DeliveredTestOutcomeDeclarationsService.php#L44): 
```
    public function __construct(
        QtiRunnerService $qtiRunnerService,
        DeliveryExecutionService $deliveryExecutionService,
        DeliveryContainerService $deliveryContainerService
    ) {
```

2. And this one, I think, is even bigger than the first one. The service itself has a [dependency on this class](https://github.com/oat-sa/extension-tao-outcome/blob/bbb231b43bb0ac052de3586fb7520649bb008110/models/Import/Service/DeliveredTestOutcomeDeclarationsService.php#L25) and in case it will not be available, we will have an issue during autoloading.

It seems to me we have a really big problem with coupling here, so as a solution, I could suggest looking at how we can replace `oat\taoDeliveryRdf\model\DeliveryContainerService` with an interface (for example `\oat\taoDelivery\model\DeliveryContainerService` or another if this one is not available) and what will not work in case this service is not available. Because dependency inversion is a technical problem that we can solve, but lack of functionality is a business issue.

Reply:
note: If you need to pass `null` in case the service is not available, you can use [this syntax](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Bundle/FrameworkBundle/Resources/config/services.php#L109) instead.

```suggestion
                    service(oat\taoDeliveryRdf\model\DeliveryContainerService::SERVICE_ID)->nullOnInvalid()
```